### PR TITLE
CONFIG/JENKINS/TEST/UCS: Fixes and adaptations for older sw versions.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10023/ for details (Mellanox internal link).

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -868,34 +878,39 @@ test_unused_env_var() {
 
 test_env_var_aliases() {
 	echo "==== Running MLX5 env var aliases test ===="
-	vars=( "TM_ENABLE" "TM_MAX_BCOPY" "TM_LIST_SIZE" "TX_MAX_BB" )
-	for var in "${vars[@]}"
-	do
-		for tl in "RC_MLX5" "DC_MLX5"
+	if [[ `./src/tools/info/ucx_info -b | grep -P 'HW_TM *1$'` ]]
+	then
+		vars=( "TM_ENABLE" "TM_MAX_BCOPY" "TM_LIST_SIZE" "TX_MAX_BB" )
```
Comment (brminich) [inline contrib/test_jenkins.sh]: `TM_MAX_BCOPY` would need to be renamed to `TM_SEG_SIZE` when #3524 is merged

File: contrib/test_jenkins.sh [hunk 2]
```diff
@@ -430,19 +431,27 @@ build_clang() {
 #
 build_gcc_latest() {
 	echo 1..1 > build_gcc_latest.tap
-	if module_load dev/gcc-latest
+    #If the gcc version on the host is older than 4.8.5, don't run
+    if (echo "4.8.5"; gcc --version | head -1 | awk '{print $3}') | sort -CV
```
Comment (brminich) [inline contrib/test_jenkins.sh]: what does it check?
```
mikhailb@jazz18 ~/wgit/ompi
$gcc --version | head -1 | awk '{print $3}'
4.8.5
mikhailb@jazz18 ~/wgit/ompi
$if (echo "0.0.1"; gcc --version | head -1 | awk '{print $3}') | sort -CV ; then echo "equal"; else echo "NOT equal"; fi
equal

```

**Comment by alinask** [comment]
@yosefe what response is required?

File: contrib/test_jenkins.sh [hunk 3]
```diff
@@ -430,19 +431,27 @@ build_clang() {
 #
 build_gcc_latest() {
 	echo 1..1 > build_gcc_latest.tap
-	if module_load dev/gcc-latest
+    #If the gcc version on the host is older than 4.8.5, don't run
```
[the remark this answers is not in the document]
Comment (alinask) [reply contrib/test_jenkins.sh]: Yes, glibc

File: contrib/test_jenkins.sh [hunk 4]
```diff
@@ -758,7 +767,7 @@ test_malloc_hooks_mpi() {
 #
 run_mpi_tests() {
 	echo "1..2" > mpi_tests.tap
-	if module_load hpcx-gcc
+	if module_load hpcx-gcc && ucx_info -v
```
[the remark this answers is not in the document]
Comment (alinask) [reply contrib/test_jenkins.sh]: It's just a way to check that the module is valid on the used host..

[the remark this answers is not in the document]
Comment (yosefe) [reply contrib/test_jenkins.sh]: need to check that mpicc/mpirun

File: config/m4/compiler.m4 [hunk 5]
```diff
@@ -316,7 +316,15 @@ AC_LANG_PUSH([C++])
...
 CXX11FLAGS="-std=c++11"
 CXXFLAGS="$CXXFLAGS $CXX11FLAGS"
-AC_COMPILE_IFELSE([AC_LANG_SOURCE([[int main() { return 0; } ]])],
+AC_COMPILE_IFELSE([AC_LANG_SOURCE([[#include <iostream>
+					#include <string>
+					int main() {
+						std::string my_str;
+						int a = 5;
+						my_str = std::to_string(a);
+						std::cout << "to_string: " << my_str << '\n';
```
[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply config/m4/compiler.m4]: this code can be optimized out by compiler, I would propose:
```
int main() { return (int)std::to_string(1).length(); }
```

[the remark this answers is not in the document]
Comment (yosefe) [reply config/m4/compiler.m4]: it doesn't matter if it's optimized by the compiler, we are checking that it passes compiler frontend stage

[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply config/m4/compiler.m4]: ok, agree

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10053/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10072/ for details (Mellanox internal link).

**Comment by alinask** [comment]
@yosefe Can you please review?

File: contrib/test_jenkins.sh [hunk 6]
```diff
@@ -430,19 +431,29 @@ build_clang() {
 #
 build_gcc_latest() {
 	echo 1..1 > build_gcc_latest.tap
-	if module_load dev/gcc-latest
+	#If the glibc version on the host is older than 2.14, don't run
+	#check the glibc version with the ldd version since it comes with glibc
```
Comment (yosefe) [inline contrib/test_jenkins.sh]: minor: pls add ref to why this is a good method to check ldd ver

File: contrib/test_jenkins.sh (hunk 1 above)
Comment (yosefe) [reply contrib/test_jenkins.sh]: lets remove TM_MAX_BCOPY

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10078/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10084/ for details (Mellanox internal link).

**Comment by alinask** [comment]
bot:mlx:retest

**Pull request merged**